config.go

```diff
@@ -492,10 +492,14 @@ type ModuleConfig interface {
 	//
 	// # Notes
 	//
-	//   - If any function doesn't exist, it is skipped. However, all functions
-	//	  that do exist are called in order.
-	//   - Some start functions may exit the module during instantiate with a
-	//	  sys.ExitError (e.g. emscripten), preventing use of exported functions.
+	//   - If a start function doesn't exist, it is skipped. However, any that
+	//     do exist are called in order.
+	//   - Start functions are not intended to be called multiple times.
+	//     Functions that should be called multiple times should be invoked
+	//     manually via api.Module's `ExportedFunction` method.
+	//   - Start functions commonly exit the module during instantiation,
+	//     preventing use of any functions later. This is the case in "wasip1",
+	//     which defines the default value "_start".
 	WithStartFunctions(...string) ModuleConfig
```

Comment:
unless it's undocumented for a reason (i.e. unintended behavior) I think it's worth mentioning that invoking this method with 0 args clears the list, which effectively disables invoking any start function (so that, for instance, you can invoke it explicitly). I don't know if it is useful since you can always `CompileModule` instead of `Instantiate` but... https://github.com/tetratelabs/wazero/issues/1561#issuecomment-1623615488

Reply:
cool I added the note about clearing here https://github.com/tetratelabs/wazero/pull/1569

in reality any serious use of wasi, as a command/cli, needs to override ModuleConfig because the only side effects you can view otherwise is the exit code.